clientd3d/ownerdrw.c

```diff
@@ -485,3 +486,21 @@ void DrawOwnerListItem(const DRAWITEMSTRUCT *lpdis, Bool selected, Bool combo)
       DrawFocusRect(lpdis->hDC, &lpdis->rcItem);
    }
 }
+/*****************************************************************************/
+/* 
+ * IsValidRarity:  checks whether a given rarity value is valid
+ */
+int IsValidRarity(int rarity) {
+   switch (rarity) {
+      case ITEM_RARITY_GRADE_NORMAL:
+      case ITEM_RARITY_GRADE_UNCOMMON:
+      case ITEM_RARITY_GRADE_RARE:
+      case ITEM_RARITY_GRADE_LEGENDARY:
+      case ITEM_RARITY_GRADE_UNIDENTIFIED:
+      case ITEM_RARITY_GRADE_CURSED:
```

Comment:
I don't know if this is bad, because we will need to update rarity grades in both proto.h and in this function which feels bad.  But I'm not sure how to validate the input more efficiently.

Reply:
Removed